elasticsearch_dsl/mapping_base.py

```diff
@@ -57,7 +58,7 @@ def field(self, name, *args, **kwargs):
         self.properties[name] = construct_field(*args, **kwargs)
         return self
 
-    def _collect_fields(self):
+    def _collect_fields(self) -> Generator[Field, None, None]:
```

Comment:
Should this be an `Iterator[Field]` then?

Reply:
Okay, sure, that works too.